lib/report_plugins/weektoweek.js

```diff
@@ -0,0 +1,353 @@
+  legend += '<td><svg width="16" height="16"><circle fill="' + dayColors[colorIdx++] + '" r="40"></circle></g></svg></td><td>Friday</td>';
+  legend += '<td><svg width="16" height="16"><circle fill="' + dayColors[colorIdx++] + '" r="40"></circle></g></svg></td><td>Saturday</td></tr>';
+  legend += '</table>';
+
+  $('#weektoweekcharts').append($(legend));
+
+  weekstoshow.forEach(function eachWeek(d) {
+    $('#weektoweekcharts').append($('<table><tr><td><div id="weektoweekchart-' + d[0] + '-' + d[d.length-1] + '"></div></td><td><div id="weektoweekstatchart-' + d[0] + '-' + d[d.length-1] + '"></td></tr></table>'));
+  });
+};
+
+weektoweek.report = function report_weektoweek(datastorage, sorteddaystoshow, options) {
+  var Nightscout = window.Nightscout;
+  var client = Nightscout.client;
+  var translate = client.translate;
+  var profile = client.sbx.data.profile;
+  var report_plugins = Nightscout.report_plugins;
+  var scaledTreatmentBG = report_plugins.utils.scaledTreatmentBG;
+  
+  var TOOLTIP_TRANS_MS = 300;
+
+  var padding = { top: 15, right: 22, bottom: 30, left: 35 };
+
+  var tddSum = 0;
+  var carbsSum = 0;
+
+  var weekstoshow = [ ];
+  weekstoshow[0] = [ ];
+  var weekIdx = 0;
+  var dayIdx = 0;
+  var reset_day = 0;
+
+  reset_day = moment(sorteddaystoshow[0]).day();
+
+  sorteddaystoshow.forEach( function eachDay(day) {
+    if ((dayIdx > 0) && (moment(day).day() === reset_day)) {
+      weekIdx += 1;
+      weekstoshow[weekIdx] = [ ];
+      weekstoshow[weekIdx][0] = day;
+      dayIdx = 1;
+    } else {
+      weekstoshow[weekIdx][dayIdx] = day;
+      dayIdx += 1;
+    }
+  });
+
+  weekstoshow = weekstoshow.map(function orderWeek(week) {
+    return _.sortBy(week);
+  });
+
+  weektoweek.prepareHtml(weekstoshow);
+
+  weekstoshow.forEach( function eachWeek(week) {
+    var sgvData = [ ];
+    var currDay = 0;
+
+    week.forEach( function eachDay(day) {
+      // console.log('Rectifying day: ' + day);
+      // console.log('First sgv: ' + datastorage[day].sgv[0].date + ' last sgv: ' + datastorage[day].sgv[datastorage[day].sgv.length - 1].date);
+      datastorage[day].sgv.forEach ( function eachSgv(sgv) {
+        var sgvDate = moment(sgv.date);
+        var sgvWeekday = moment(sgv.date).day();
+        var sgvColor = dayColors[sgvWeekday];
+
+        if (sgv.color === 'gray') {
+          sgvColor = sgv.color;
+        }
+
+        sgvData.push( {
+          'color': sgvColor
+          , 'date': moment(sgv.date).subtract(currDay, 'days').toDate()
+          , 'filtered': sgv.filtered
+          , 'mills': sgv.mills - currDay * 24*60*60000
+          , 'noise': sgv.noise
+          , 'sgv': sgv.sgv
+          , 'type': sgv.type
+          , 'unfiltered': sgv.unfiltered
+          , 'y': sgv.y
+        });
+      });
+
+      // console.log('First sgv: ' + sgvData[0].date + ' last sgv: ' + sgvData[sgvData.length - 1].date);
```

Comment:
can you remove logging code and add a unit test for this report?

Reply:
@PieterGit, a unit test is implemented and the logging code is removed.